src/core/src/runtime/tensor.cpp

```diff
@@ -152,14 +153,11 @@ ov::Shape calc_static_shape_for_file(const std::filesystem::path& file_name,
                     " bytes");
     auto elements_to_read = file_size_to_read * 8 / element_type.bitwidth();
 
-    auto new_dimension_size = elements_to_read / slice_size;
-    OPENVINO_ASSERT(new_dimension_size * slice_size == elements_to_read,
-                    "Cannot fit file size into requested PartialShape");
-
-    OPENVINO_ASSERT(dynamic_dimension.compatible(new_dimension_size),
+    auto new_dimension = elements_to_read / slice_size;
+    OPENVINO_ASSERT((new_dimension * slice_size == elements_to_read) && dynamic_dimension.compatible(new_dimension),
```

Comment:
There is no need to do calculation to check if division is correct. The `ov::Dimenion` div operator produce invalid interval is dimension cannot be divided.
There is helper to check interval ```ov::util::dim::is_empty(new_dimension) ``` but Dimenion with invalid interval should not be compatible with any other.

Reply:
done